### ISSUE TITLE
linter: vitest/no-conditional-in-test false positive with nullish coalescing

### ISSUE THREAD

**Issue by Huliiiiii** [oxc-project/oxc#22496]
### What version of Oxlint are you using?

1.65.0

### What command did you run?

oxlint --type-aware --type-check --report-unused-disable-directives

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
	"plugins": ["vitest"],
	"rules": {
		"vitest/no-conditional-in-test": "warn"
	}
}
```

### What happened?

Minimal reproduction:

```ts
import { expect, it } from "vitest"

declare const expected: number | undefined

it("false positive", () => {
	expect(1).toBe(expected ?? 1)
})
```

Oxlint reports:
```
vitest(no-conditional-in-test): Avoid having conditionals in tests.
```

**Comment by camc314** [comment]
Hi, thanks for reporting this issue.

I can see this behaviour doesn't align with the upstream plugin, however I think this is a better behaviour.

relying on the logical expression to satisfy the test could results in flaky tests.

If `expected` is always defined, the `?? 1` is unnecessary.
If `expected` is never defined, the `?? 1` is always required, so `expected` is never really tested.
If `expected` is only sometimes defined, that indicates that the tests are flaky or are written in an unconventional way. Relying on the fallback is not a good decision as you are no longer testing the expected value


If `undefined` AND `1` are valid for `expected`, rewrite the expect assertion as below to avoid the conditional.
```
expect(expected).toBeOneOf(["1", null, undefined']);
```

Closing as not planned - thanks!

**Comment by Huliiiiii** [comment]
@camc314 
Just because you think this is better behavior doesn't mean it can be inconsistent with the description and the original plugin. Oxlint can add another lint, but it doesn't have to change the meaning of the original lint.

In my case, this is the behavior of normalizing an nullable array that the API returns to an array. 

The original code was like:
```
fc.property(inputArb, (input) => {
  const result = parse(input)
  if (input.arrayField) {
    expect(result.arrayField).toHaveLength(input.arrayField.length)
  } else {
    expect(result.arrayField).toEquals([])
  }
}
```

I don't think this is a bad test.

The example is just me reducing it to a minimal, reproducible form, and I think there's an error in the implementation regarding which parts of the code contain the conditional expression.

**Comment by camc314** [comment]
> The original code was like:

Ok but these are two different issues.

In this case you're using an `if` stmt, but in the original issue you're using nullish coalescing?

how did you get from an if stmt to a logical expression

**Comment by Huliiiiii** [comment]
Sorry, I seem to have mixed up `no-conditional-in-test` and `no-conditional-tests`.

**Comment by camc314** [comment]
all good - given that is the issue still valid?

please make a new one if needed.

thanks

**Comment by Huliiiiii** [comment]
However, the issue persists. The [original plugin](https://github.com/vitest-dev/eslint-plugin-vitest/blob/789966ecb8ffae10aee8e59d843f85a6dd00f031/src/rules/no-conditional-in-test.ts) does not report `SwitchStatement`, `ConditionalExpression`, and `LogicalExpression`.

https://github.com/oxc-project/oxc/blob/610f4c7b0bdac95148373fab8798e8fcbecec820/crates/oxc_linter/src/rules/vitest/no_conditional_in_test.rs#L16-L23

```ts
create(context) {
  return {
    IfStatement(node) {
      if (
        node.parent?.parent?.parent?.type === 'CallExpression' &&
        isTypeOfVitestFnCall(node.parent?.parent?.parent, context, [
          'test',
          'it',
        ])
      ) {
        context.report({
          messageId: 'noConditionalInTest',
          node,
        })
      }
    },
  }
},
```

**Comment by Huliiiiii** [comment]
Jest's [implementations](https://github.com/jest-community/eslint-plugin-jest/blob/b99680974af3a1c53230e15fc5bef84485edd09c/src/rules/no-conditional-in-test.ts#L4) differs from the vitest plugin and is closer to oxlint, but jest has the allowOptionalChaining option.

**Comment by camc314** [comment]
i've reopened the isssue, but i'm going to hold off fixing/closing it until https://github.com/vitest-dev/eslint-plugin-vitest/issues/900 is either closed as won't fix or resolved.

**Comment by Huliiiiii** [comment]
I checked the history, and the initial implementation was consistent with jest vitest-dev/eslint-plugin-vitest#69. The implementation was later updated due to issue vitest-dev/eslint-plugin-vitest#358 (unable to reproduce using oxlint). 

Therefore, I think this issue can be closed, because the current implementation in oxlint is correct.
